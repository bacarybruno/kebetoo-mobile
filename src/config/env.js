/* eslint-disable no-undef */

__DEV__ = false

// TODO: use env variables
export default {
  apiBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.herokuapp.com',
  assetsBaseUrl: __DEV__ ? 'http://localhost:1337' : 'https://kebetoo.s3.eu-west-3.amazonaws.com',
}
