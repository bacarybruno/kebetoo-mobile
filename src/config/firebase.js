import { utils } from '@react-native-firebase/app';
import analytics from '@react-native-firebase/analytics';

if (utils().isRunningInTestLab) {
  // do not track analytics for firebase test lab
  // used by google play to run test on publish
  analytics().setAnalyticsCollectionEnabled(false);
}
