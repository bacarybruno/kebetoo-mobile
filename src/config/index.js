export { default as env } from './env';
export { default as strings } from './strings';
