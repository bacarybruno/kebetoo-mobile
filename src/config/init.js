import { GoogleSignin } from '@react-native-community/google-signin'
import dayjs from 'dayjs'
import relativeTime from 'dayjs/plugin/relativeTime'

GoogleSignin.configure({
  scopes: ['email', 'profile', 'openid'],
  webClientId: '355888072499-spu5d874qrfb6tfs3pq195cjf6ij56go.apps.googleusercontent.com',
})

dayjs.extend(relativeTime)
