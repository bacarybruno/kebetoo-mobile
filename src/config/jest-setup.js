/* eslint-disable import/no-extraneous-dependencies */
import { NativeModules } from 'react-native';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MockDate from 'mockdate';
import configureStore from 'redux-mock-store';
import 'react-native-gesture-handler/jestSetup';
import { renderHook } from '@testing-library/react-hooks';
import auth from '@react-native-firebase/auth';
import { SafeAreaContext } from '@app/shared/contexts';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { PortalProvider } from '@gorhom/portal';

// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

// wix/react-native-keyboard-input
NativeModules.KeyboardTrackingViewTempManager = {
  KeyboardTrackingScrollBehaviorNone: 'NONE',
  KeyboardTrackingScrollBehaviorScrollToBottomInvertedOnly: 'SCROLL_TO_BOTTOM_INVERTED_ONLY',
  KeyboardTrackingScrollBehaviorFixedOffset: 'FIXED_OFFSET',
};

NativeModules.RealPathUtils = {
  getOriginalFilePath: jest.fn(),
};

const mockedStoreState = {
  postsReducer: {
    posts: [],
    authors: [],
  },
  notificationsReducer: {
    notifications: [],
  },
  userReducer: {
    profile: {
      ...auth().currentUser,
      isLoggedIn: true,
    },
  },
  appReducer: {
    theme: 'system',
    locale: 'fr',
  },
};
// helper to setup unit tests
/* eslint-disable comma-dangle */
const setupTest = (WrappedComponent, renderFn = TestRenderer.create) => {
  let wrapper = null;
  return (defaultProps = {}) => (additionalProps = {}) => {
    const {
      store, __storeState__ = mockedStoreState, disconnectNavigation, ...props
    } = defaultProps;
    const propsWithArgs = {
      ...props,
      ...additionalProps,
    };
    const mockStore = configureStore();
    const safeAreaCtxValue = {
      updateTopSafeAreaColor: jest.fn(),
      updateBottomSafeAreaColor: jest.fn(),
      resetStatusBars: jest.fn(),
    };
    const Stack = createStackNavigator();
    const Component = (
      <SafeAreaContext.Provider value={safeAreaCtxValue}>
        <Provider store={store || mockStore(__storeState__)}>
          <PortalProvider>
            {disconnectNavigation
              ? <WrappedComponent {...propsWithArgs} />
              : (
                <NavigationContainer>
                  <Stack.Navigator>
                    <Stack.Screen name="jest">
                      {() => <WrappedComponent {...propsWithArgs} />}
                    </Stack.Screen>
                  </Stack.Navigator>
                </NavigationContainer>
              )}
          </PortalProvider>
        </Provider>
      </SafeAreaContext.Provider>
    );
    wrapper = renderFn(Component);
    return { wrapper, props: propsWithArgs };
  };
};

export const setupHook = (useHook, ...props) => {
  const mockStore = configureStore();
  const store = mockStore(mockedStoreState);
  const wrapper = ({ children }) => (
    <Provider store={store}>
      {children}
    </Provider>
  );
  const rendered = renderHook(() => useHook(...props), {
    wrapper,
  });
  return rendered;
};

/**
 * Explicit mocks
 * Create mocks here only if they can't be created on the __mocks__ folder
 */

jest.useFakeTimers();
jest.setTimeout(30000);

// react-navigation
const mockedNavigate = jest.fn();
const mockedSetOptions = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockedNavigate,
    setOptions: mockedSetOptions,
    addListener: jest.fn(),
  }),
  useRoute: () => ({
    params: {},
  }),
  useFocusEffect: (cb) => cb()
}));
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => { };
  Reanimated.default.addWhitelistedUIProps = () => { };

  return Reanimated;
});

// react-native-screens
jest.mock('react-native-screens', () => {
  const Screens = jest.requireActual('react-native-screens');
  Screens.enableScreens = () => { };
  return Screens;
});

// date
MockDate.set(new Date(Date.parse('2020-06-25T13:30:00+02:00')));

export default setupTest;
