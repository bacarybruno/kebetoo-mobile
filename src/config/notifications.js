import { Platform } from 'react-native'
import PushNotification from 'react-native-push-notification'
import AsyncStorage from '@react-native-community/async-storage'
import messaging from '@react-native-firebase/messaging'

import { getNotificationMessage, getNotificationTitle } from '@app/features/notifications/containers'

PushNotification.createChannel({
  channelId: 'kbt',
  channelName: 'Kebetoo Channel',
  channelDescription: 'Local notifications sent by Kebetoo',
  playSound: false,
})

const formatNotificationMessage = ({ name, message }) => `${name} ${message}`

PushNotification.configure({
  requestPermissions: Platform.OS === 'ios',
})

// handle background notifications
messaging().setBackgroundMessageHandler(async (remoteMessage) => {
  const bgNotifications = JSON.parse(await AsyncStorage.getItem('backgroundNotifications')) || []
  bgNotifications.push(remoteMessage)
  PushNotification.setApplicationIconBadgeNumber(bgNotifications.length)
  await AsyncStorage.setItem('backgroundNotifications', JSON.stringify(bgNotifications))

  const notificationTitle = getNotificationTitle(remoteMessage)
  if (!notificationTitle && !remoteMessage.data.title) return

  PushNotification.localNotification({
    channelId: 'kbt',
    title: notificationTitle
      ? formatNotificationMessage(notificationTitle)
      : remoteMessage.data.title,
    message: getNotificationMessage(remoteMessage) || remoteMessage.data.message,
    smallIcon: 'ic_stat_ic_notification',
  })
})
