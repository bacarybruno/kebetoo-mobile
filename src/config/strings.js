import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
