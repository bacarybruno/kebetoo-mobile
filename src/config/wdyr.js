import * as React from 'react';

if (process.env.NODE_ENV === 'development') {
  // eslint-disable-next-line import/no-extraneous-dependencies
  const whyDidYouRender = require('@welldone-software/why-did-you-render');
  const ReactRedux = require('react-redux');
  whyDidYouRender(React, {
    runtime: 'automatic',
    trackAllPureComponents: true,
    trackExtraHooks: [
      [ReactRedux, 'useSelector'],
    ],
  });
}
