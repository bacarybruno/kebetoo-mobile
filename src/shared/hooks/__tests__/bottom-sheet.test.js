import { setupHook } from '@app/config/jest-setup'
import { useActionSheet } from '@expo/react-native-action-sheet'
import useBottomSheet from '../bottom-sheet'

jest.mock('@expo/react-native-action-sheet', () => ({
  useActionSheet: jest.fn().mockReturnValue({
    showActionSheetWithOptions: jest.fn().mockImplementation((props, callback) => callback(0)),
  }),
}))

beforeEach(jest.clearAllMocks)

const givenBottomSheet = () => {
  const rendered = setupHook(useBottomSheet)
  return { bottomSheet: rendered.result.current, rerender: rendered.rerender }
}

it('showManagePostsOptions', async () => {
  const { bottomSheet } = givenBottomSheet()
  await bottomSheet.showManagePostsOptions()
  expect(useActionSheet().showActionSheetWithOptions.mock.calls[0]).toMatchSnapshot()
})

it('showSharePostOptions', async () => {
  const { bottomSheet } = givenBottomSheet()
  await bottomSheet.showSharePostOptions()
  expect(useActionSheet().showActionSheetWithOptions.mock.calls[0]).toMatchSnapshot()
})

it('showAppearanceOptions', async () => {
  const { bottomSheet } = givenBottomSheet()
  await bottomSheet.showAppearanceOptions()
  expect(useActionSheet().showActionSheetWithOptions.mock.calls[0]).toMatchSnapshot()
})

it('showAvatarOptions', async () => {
  const { bottomSheet } = givenBottomSheet()
  await bottomSheet.showAvatarOptions()
  expect(useActionSheet().showActionSheetWithOptions.mock.calls[0]).toMatchSnapshot()
})

it('showFeedPostsOptions', async () => {
  const { bottomSheet } = givenBottomSheet()
  await bottomSheet.showFeedPostsOptions({
    author: {
      displayName: 'Bruno',
    },
  })
  expect(useActionSheet().showActionSheetWithOptions.mock.calls[0]).toMatchSnapshot()
})
