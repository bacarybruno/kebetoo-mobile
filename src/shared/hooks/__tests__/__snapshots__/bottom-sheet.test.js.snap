// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showAppearanceOptions 1`] = `
Array [
  Object {
    "cancelButtonIndex": 2,
    "containerStyle": Object {
      "backgroundColor": "#F5F5F5",
    },
    "destructiveButtonIndex": 1,
    "icons": Array [
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-create"
        size={24}
      />,
      <_default
        color="rgba(255, 45, 85, 255)"
        name="ios-trash"
        size={24}
      />,
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-close"
        size={24}
      />,
    ],
    "options": Array [
      "Edit post",
      "Delete post",
      "Cancel",
    ],
    "textStyle": Object {
      "color": "rgba(0,0,0,0.87)",
    },
    "title": "Actions",
    "titleTextStyle": Object {
      "color": "rgba(0,0,0,0.54)",
    },
  },
  [Function],
]
`;

exports[`showAvatarOptions 1`] = `
Array [
  Object {
    "cancelButtonIndex": 2,
    "containerStyle": Object {
      "backgroundColor": "#F5F5F5",
    },
    "destructiveButtonIndex": 1,
    "icons": Array [
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-create"
        size={24}
      />,
      <_default
        color="rgba(255, 45, 85, 255)"
        name="ios-trash"
        size={24}
      />,
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-close"
        size={24}
      />,
    ],
    "options": Array [
      "Edit post",
      "Delete post",
      "Cancel",
    ],
    "textStyle": Object {
      "color": "rgba(0,0,0,0.87)",
    },
    "title": "Actions",
    "titleTextStyle": Object {
      "color": "rgba(0,0,0,0.54)",
    },
  },
  [Function],
]
`;

exports[`showFeedPostsOptions 1`] = `
Array [
  Object {
    "cancelButtonIndex": 2,
    "containerStyle": Object {
      "backgroundColor": "#F5F5F5",
    },
    "destructiveButtonIndex": 1,
    "icons": Array [
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-create"
        size={24}
      />,
      <_default
        color="rgba(255, 45, 85, 255)"
        name="ios-trash"
        size={24}
      />,
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-close"
        size={24}
      />,
    ],
    "options": Array [
      "Edit post",
      "Delete post",
      "Cancel",
    ],
    "textStyle": Object {
      "color": "rgba(0,0,0,0.87)",
    },
    "title": "Actions",
    "titleTextStyle": Object {
      "color": "rgba(0,0,0,0.54)",
    },
  },
  [Function],
]
`;

exports[`showManagePostsOptions 1`] = `
Array [
  Object {
    "cancelButtonIndex": 2,
    "containerStyle": Object {
      "backgroundColor": "#F5F5F5",
    },
    "destructiveButtonIndex": 1,
    "icons": Array [
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-create"
        size={24}
      />,
      <_default
        color="rgba(255, 45, 85, 255)"
        name="ios-trash"
        size={24}
      />,
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-close"
        size={24}
      />,
    ],
    "options": Array [
      "Edit post",
      "Delete post",
      "Cancel",
    ],
    "textStyle": Object {
      "color": "rgba(0,0,0,0.87)",
    },
    "title": "Actions",
    "titleTextStyle": Object {
      "color": "rgba(0,0,0,0.54)",
    },
  },
  [Function],
]
`;

exports[`showSharePostOptions 1`] = `
Array [
  Object {
    "cancelButtonIndex": 2,
    "containerStyle": Object {
      "backgroundColor": "#F5F5F5",
    },
    "destructiveButtonIndex": 1,
    "icons": Array [
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-create"
        size={24}
      />,
      <_default
        color="rgba(255, 45, 85, 255)"
        name="ios-trash"
        size={24}
      />,
      <_default
        color="rgba(0,0,0,0.87)"
        name="md-close"
        size={24}
      />,
    ],
    "options": Array [
      "Edit post",
      "Delete post",
      "Cancel",
    ],
    "textStyle": Object {
      "color": "rgba(0,0,0,0.87)",
    },
    "title": "Actions",
    "titleTextStyle": Object {
      "color": "rgba(0,0,0,0.54)",
    },
  },
  [Function],
]
`;
